src/bidiMapper/domains/script/scriptEvaluator.ts

```diff
@@ -17,16 +17,74 @@
 import {Protocol} from 'devtools-protocol';
 
 import {CommonDataTypes, Message, Script} from '../../../protocol/protocol.js';
+import {IEventManager} from '../events/EventManager.js';
 
 import {Realm} from './realm.js';
 
+async function longPoll(
+  channel: Script.Channel,
+  channelHandle: string | undefined,
+  eventManager: IEventManager,
+  realm: Realm
+) {
+  const channelId = channel.value.channel;
+
+  // TODO(#294): Remove this loop after the realm is destroyed.
+  // Rely on the CDP throwing exception in such a case.
+  for (;;) {
+    const message = await realm.cdpClient.sendCommand(
+      'Runtime.callFunctionOn',
+      {
+        functionDeclaration: String(
+          async (channelHandle: {
+            getMessage: () => Promise<unknown>;
+            sendMessage: (message: string) => void;
```

Comment:
NIT: Listener doesn't need to know about `sendMessage` method.
```suggestion
```

Reply:
Done